src/go/rpk/pkg/cli/cmd/redpanda/admin/selftest/selftest.go

```diff
@@ -0,0 +1,344 @@
+	test := SelfTestNetcheckParameters{
+		Name:        "8K Network Throughput Test",
+		RequestSize: (2 << 12), // 8192 bytes
+		DurationMs:  int64(duration),
+		Parallelism: 10,
+		Type:        netcheckTagIdentifier,
+	}
+	return []SelfTestNetcheckParameters{test}
+}
+
+func DefaultConnectivityTest() SelfTestNetcheckParameters {
+	test := SelfTestNetcheckParameters{
+		Name:        "Network Connectivity Test",
+		RequestSize: 512,
+		DurationMs:  1000, // 500ms
+		Parallelism: 1,
+		Type:        netcheckTagIdentifier,
+	}
+	return test
+}
+
+func DefaultTests(durationDisk int, durationNet int) []interface{} {
+	tests := make([]interface{}, 0)
+	for _, t := range DefaultDiskcheckTests(durationDisk) {
+		tests = append(tests, t)
+	}
+	for _, t := range DefaultNetcheckTests(durationNet) {
+		tests = append(tests, t)
+	}
+	return tests
+}
+
+// Assembles certain types of pre-canned tests depending on user input
+// 1. onlyDisk - Only runs the throughput/latency disk benchmarks
+// 2. onlyNetwork - Only runs the network benchmark
+// 3. onlyConn - Runs the network benchmark with short duration - no metrics
+// 4. All false - Runs the default which is the combination of option 1 & 2.
+func AssembleTests(onlyDisk bool, onlyNetwork bool, onlyConn bool, durationDisk int, durationNet int) []interface{} {
+	tests := make([]interface{}, 0)
+	if onlyDisk {
+		for _, t := range DefaultDiskcheckTests(durationDisk) {
+			tests = append(tests, t)
+		}
+	} else if onlyNetwork {
+		for _, t := range DefaultNetcheckTests(durationNet) {
+			tests = append(tests, t)
+		}
+	} else if onlyConn {
+		tests = append(tests, DefaultConnectivityTest())
+	} else {
+		tests = DefaultTests(durationDisk, durationNet)
+	}
+	return tests
+}
+
+func NewCommand(fs afero.Fs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "self-test",
+		Short: "Start, stop and query redpanda self-test runs through the admin listener",
+		Args:  cobra.ExactArgs(0),
+	}
+	cmd.AddCommand(
+		newStartSelfTest(fs),
+		newStopSelfTest(fs),
+		newQuerySelfTest(fs),
+	)
+	return cmd
+}
+
+func newStartSelfTest(fs afero.Fs) *cobra.Command {
+	var (
+		yes              bool
+		diskDurationMs   int
+		netDurationMs    int
+		onNodes          []int
+		onlyDisk         bool
+		onlyNetwork      bool
+		onlyConnectivity bool
+	)
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Starts a new self test run",
+		Args:  cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			// Load config settings
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			// Warn user before continuing, proceed only via explicit signal
+			if !yes {
+				fmt.Print("Invoking the redpanda self tests may put excessive pressure on your system. It attempts to benchmark disk and network hardware to find their max throughputs. Its advised to not start the test if there are already large workloads running on the system. To continue type [Y/y]:")
```

Comment:
We have a package method for confirmation prompts: out.Confirm()

See an example usage here https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/cmd/acl/delete.go#L75